apisix/plugins/cors.lua

```diff
@@ -179,8 +193,49 @@ function _M.header_filter(conf, ctx)
         end
     end
 
-    ctx.cors_allow_origins = allow_origins
-    set_cors_headers(conf, ctx)
+    return allow_origins
+end
+
+local function process_with_allow_origins_by_regex(conf, ctx)
+    if conf.allow_origins_by_regex == nil  or next(conf.allow_origins_by_regex) == nil then
+        return
+    end
+    local req_origin = core.request.header(ctx, "Origin")
```

Comment:
Check the existence of `Origin` header is required.

Reply:
how to do that? lua  needs to set values into array, https://stackoverflow.com/questions/28502501/how-to-declare-an-array-with-x-elements-in-lua